### DISCUSSION TITLE
Avoid Repeated "poetry run"

### DISCUSSION THREAD

**Discussion by cjuracek** [python-poetry/poetry#6894] [Q&A] [answered]
Hello all,

When I am in a poetry project directory, I almost always want to run commands from within the project's virtualenv. Thus I find it slightly tedious to constantly type `poetry run` before every command. Compare to a virtualenv workflow _without_ poetry, where the environment is activated once and then no further action needs to be taken to have commands run in that virtualenv. Is there any similar workflow in Poetry?

Thanks!

**Comment by neersighted** [comment, marked as answer]
`poetry shell` or `source $(poetry env info --path)/bin/activate` -- overall it's "just" another virtual environment.

**Note:** Since Poetry 2.0.0, the `poetry shell` command is only available as a plugin. You can use `$(poetry env activate)` instead.

**Reply by cjuracek** [reply]
Feels obvious now that you've said it. I was not aware of `poetry shell`, but I see this being a mainstay of my workflow now.

Thanks for the quick response!

**Comment by TheOutdoorProgrammer** [comment]
If anyone is coming across this in 2025, I believe the correct answer is now `$(poetry env activate)` since the `poetry shell` command has been removed.

**Reply by ritzvincentvculanag** [reply]
Nice one!
